src/main/java/org/opentripplanner/routing/trippattern/TripTimes.java

```diff
@@ -474,6 +478,66 @@ public boolean tripAcceptable(final State state0, final int stopIndex) {
         return true;
     }
 
+    /**
+     * Returns true if the current state chain contains a banned trip, or banned trip/stop index or banned trip sequence
+     */
+    public boolean tripOrTripSequenceIsBanned(State state0, final int stopIndex) {
+        final RoutingRequest options = state0.getOptions();
+
+        // check if an exact trip ID is banned
+        final BannedStopSet banned = options.bannedTrips.get(trip.getId());
+        if (banned != null && banned.contains(stopIndex)) {
```

Comment:
This check and its comment is confusing to me, but that might just be due to ignorance. If we're checking for a particular stop sequence, how does that correspond to an "exact trip ID". This might just need some better comments.

Reply:
Also, what about the case where `banned` is empty. Doesn't that equate to the case where all stops along the trip are banned?

https://github.com/ibi-group/OpenTripPlanner/blob/4da23acf60543a9a6a532c8bafb4fc00996ade1b/src/main/java/org/opentripplanner/routing/request/BannedStopSet.java#L8

Reply:
I changed the comment and variable name to what Binh suggested in 6adec27faa2e744f94cd91e4ae4d337a5c25c7c7. There isn't any code path for a BannedStopSet to exist that is empty. If there aren't any stop sequences included in the request, then the ALL instance is used and the overridden contains method will detect this and return true for any stop sequence.